db/database.go

```diff
@@ -1071,151 +1074,189 @@ func (db *Database) UpdateAllDocChannels() (int, error) {
 
 	base.Infof(base.KeyAll, "Recomputing document channels...")
 
-	results, err := db.QueryResync()
-	if err != nil {
-		return 0, err
-	}
-
 	// We are about to alter documents without updating their sequence numbers, which would
 	// really confuse the changeCache, so turn it off until we're done:
 	db.changeCache.EnableChannelIndexing(false)
 	defer db.changeCache.EnableChannelIndexing(true)
 
-	err = db.changeCache.Clear()
+	err := db.changeCache.Clear()
 	if err != nil {
 		return 0, err
 	}
 
 	base.Infof(base.KeyAll, "Re-running sync function on all documents...")
-	changeCount := 0
-	docCount := 0
-
-	var importRow QueryIdRow
-	for results.Next(&importRow) {
-		docid := importRow.Id
-		key := realDocID(docid)
-
-		docCount++
-		documentUpdateFunc := func(doc *Document) (updatedDoc *Document, shouldUpdate bool, updatedExpiry *uint32, err error) {
-			imported := false
-			if !doc.HasValidSyncData() {
-				// This is a document not known to the sync gateway. Ignore it:
-				return nil, false, nil, base.ErrUpdateCancel
-			} else {
-				base.Debugf(base.KeyCRUD, "\tRe-syncing document %q", base.UD(docid))
-			}
 
-			// Run the sync fn over each current/leaf revision, in case there are conflicts:
-			changed := 0
-			doc.History.forEachLeaf(func(rev *RevInfo) {
-				bodyBytes, _, err := db.get1xRevFromDoc(doc, rev.ID, false)
-				if err != nil {
-					base.Warnf("Error getting rev from doc %s/%s %s", base.UD(docid), rev.ID, err)
-				}
-				var body Body
-				if err := body.Unmarshal(bodyBytes); err != nil {
-					base.Warnf("Error unmarshalling body %s/%s for sync function %s", base.UD(docid), rev.ID, err)
-					return
+	queryLimit := db.Options.ResyncQueryLimit
+	startSeq := uint64(0)
+	endSeq, err := db.sequences.lastSequence()
```

Comment:
I think this should use getSequence(), and not lastSequence(), to avoid scenarios where this node has an unused, unreleased sequence that's lower than the highest sequence written to the bucket.  It's an unlikely race, given that the db needs to be taken offline prior to resync running, but would still be a more defensive approach, particularly since we don't care about the performance overhead of _sync:seq retrieval in the context of _resync.